<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>SMART pipeline Manual &mdash; SMART 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="SMART 1.0 documentation" href="index.html" />
    <link rel="prev" title="SMART Quick Start Guide" href="QUICKSTART.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="QUICKSTART.html" title="SMART Quick Start Guide"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">SMART 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="smart-pipeline-manual">
<h1>SMART pipeline Manual<a class="headerlink" href="#smart-pipeline-manual" title="Permalink to this headline">¶</a></h1>
<div class="section" id="what-is-the-smart-pipeline">
<h2>What is the SMART pipeline ?<a class="headerlink" href="#what-is-the-smart-pipeline" title="Permalink to this headline">¶</a></h2>
<p>SMART was set up to process PolyA sequencing data from aligned sequencing reads.
It includes the peak detection, filtering, and annotation.
If you use it, please cite :
<em>Boldina et al. 2015</em></p>
</div>
<div class="section" id="how-to-install-it">
<h2>How to install it ?<a class="headerlink" href="#how-to-install-it" title="Permalink to this headline">¶</a></h2>
<p>The following dependancies are required :
*
* R with the <em>RColorBrewer</em>, <em>ggplot2</em>, <em>rtracklayer</em>, and <em>genomicRanges</em> packages
* Samtools (&gt;0.1.18)
* BEDTools (&gt;2.17.0)</p>
<p>To install the SMART pipeline, simply extract the archive.</p>
<div class="highlight-guess"><div class="highlight"><pre>tar -zxvf smart_1.0.0.tar.gz
cd smart_1.0.0
</pre></div>
</div>
<p>Then, edit the <em>config.txt</em> file and manually defined the paths to dependencies.</p>
<table border="1" class="docutils">
<colgroup>
<col width="20%" />
<col width="80%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head" colspan="2">SYSTEM CONFIGURATION</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>SAMTOOLS_PATH</td>
<td>Full path to the samtools installation directory (&gt;0.1.18)</td>
</tr>
<tr class="row-odd"><td>BEDTOOLS_PATH</td>
<td>Full path to the BEDTools installation directory (&gt;2.17.0)</td>
</tr>
<tr class="row-even"><td>R_PATH</td>
<td>Full path to the R installation directory</td>
</tr>
<tr class="row-odd"><td>PYTHON_PATH</td>
<td>Full path to the python installation directory</td>
</tr>
<tr class="row-even"><td>AWK_PATH</td>
<td>Full path to the awk utility</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="annotation-files">
<h2>Annotation Files<a class="headerlink" href="#annotation-files" title="Permalink to this headline">¶</a></h2>
<p>The pipeline is using a couple of annotation files with gene annotation and last exons information. These files are based on UCSC Refseq gene.
This file can be downloaded from the <a class="reference external" href="https://genome.ucsc.edu/cgi-bin/hgTables">UCSC TableBrowser website</a> and should look like this :</p>
<div class="highlight-python"><div class="highlight"><pre>#bin  name    chrom   strand  txStart txEnd   cdsStart
cdsEnd        exonCount       exonStarts
exonEnds      score   name2   cdsStartStat    cdsEndStat      exonFrames
0     NM_001195025    chr1    +       134212701       134230065       134212806
      134228958       8       134212701,134221529,134222782,134224273,134224707,
      134226534,134227135,134227897,  134213049,134221650,134222806,134224425,
      134224773,134226654,134227268,134230065,        0       Nuak2   cmpl
      cmpl    0,0,1,1,0,0,0,1,
0     NM_028778       chr1    +       134212701       134230065       134212806
      134228958       7       134212701,134221529,134224273,134224707,134226534,
      134227135,134227897,    134213049,134221650,134224425,134224773,134226654,
      134227268,134230065,    0       Nuak2   cmpl    cmpl    0,0,1,0,0,0,1,
</pre></div>
</div>
<p>In order to generate all required annotation files, please set the ANNOT_DIR, ORG and UCSC_EXPORT in the configuration file</p>
<div class="highlight-guess"><div class="highlight"><pre>BUILD_ANNOT=1
ORG=mm9
UCSC_EXPORT=refseq_export_mm9.csv
</pre></div>
</div>
<p>The pipeline will start by creating the annotation files in the forder ANNOT_DIR/ORG/
The annotation only have to be generated once. Mouse annotation are provided with the pipeline.</p>
<p>In addition, SMART will look for known polyA signal in the detected peaks and flanking regions.
This list of motif is defined in the <em>polyA_signal.csv</em> file and can be edited.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">AATAAA</span>
<span class="n">ATTAAA</span>
<span class="n">AGTAAA</span>
<span class="n">TATAAA</span>
<span class="n">CATAAA</span>
<span class="n">AAGAAA</span>
<span class="n">GATAAA</span>
<span class="n">AATATA</span>
<span class="n">AATGAA</span>
<span class="n">TTTAAA</span>
<span class="n">AATACA</span>
<span class="n">AAAAAG</span>
<span class="n">ACTAAA</span>
<span class="n">AATAGA</span>
</pre></div>
</div>
</div>
<div class="section" id="how-to-use-it">
<h2>How to use it ?<a class="headerlink" href="#how-to-use-it" title="Permalink to this headline">¶</a></h2>
<p>SMART can be used both for a single sample or for a list of samples.
In the case of sample list, peaks detected in all samples are merged before annotation.
In oder to use the pipeline, please set up the configuration according to your analysis, and run the following command :</p>
<div class="highlight-guess"><div class="highlight"><pre>/bin/smart -c CONFIG [-i INPUT_BAM] [-l INPUT_LIST] -o OUTPUT_DIR
</pre></div>
</div>
</div>
<div class="section" id="id1">
<h2>How to use it ?<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<ol class="arabic simple">
<li>Copy and edit the configuration file <em>&#8216;config.txt&#8217;</em> in your local folder. The &#8216;[&#8216; options are optional.</li>
</ol>
<table border="1" class="docutils">
<colgroup>
<col width="16%" />
<col width="84%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head" colspan="2">GENE ANNOTATIONS</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>BUILD_ANNOT</td>
<td>0/1 - Run the annotation builder</td>
</tr>
<tr class="row-odd"><td>ORG</td>
<td>Organism</td>
</tr>
<tr class="row-even"><td>UCSC_EXPORT</td>
<td>UCSC reference to build the annotation (i.e. refseq_export_mm9.csv)</td>
</tr>
<tr class="row-odd"><td>POLYA_MOTIF</td>
<td>List of polyA annotation signal (i.e annotation/polyA_signal.csv)</td>
</tr>
</tbody>
</table>
<hr class="docutils" />
<table border="1" class="docutils">
<colgroup>
<col width="24%" />
<col width="76%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head" colspan="2">PEAK DETECTION</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>MIN_MAPQ</td>
<td>Minimum reads mapping quality (default: <em>20</em>)</td>
</tr>
<tr class="row-odd"><td>MAX_DIST_MERGE</td>
<td>Maximum distance between reads to be merged as a peak (default: <em>170</em>)</td>
</tr>
<tr class="row-even"><td>MIN_NB_READS_PER_PEAK</td>
<td>Minimum number of reads per peads (default: <em>5</em></td>
</tr>
</tbody>
</table>
<hr class="docutils" />
<table border="1" class="docutils">
<colgroup>
<col width="24%" />
<col width="76%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head" colspan="2">PEAK FILTERING</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>NB_STRETCH_POLYA</td>
<td>Window length to look for polyA stretch (default: <em>9</em>)</td>
</tr>
<tr class="row-odd"><td>MISM</td>
<td>Number of non-A base allowed in the NB_STRETCH_POLYA window (default: <em>1</em>)</td>
</tr>
<tr class="row-even"><td>NB_STRETCH_CONSECUTIVE</td>
<td>Minimum size of A stretch (default: <em>6</em>)</td>
</tr>
<tr class="row-odd"><td>WINSIZE_DOWN</td>
<td>Window size downstream the peaks (default: <em>150</em>)</td>
</tr>
<tr class="row-even"><td>WINSIZE_UP</td>
<td>Window size upstream the peaks (default: <em>50</em>)</td>
</tr>
<tr class="row-odd"><td>KEEP_LE_PEAKS</td>
<td>Always keep peaks in gene&#8217;s last exon (0/1, default: <em>1</em>)</td>
</tr>
</tbody>
</table>
<hr class="docutils" />
<table border="1" class="docutils">
<colgroup>
<col width="15%" />
<col width="85%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head" colspan="2">ANNOTATION</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>MIN_LE_OV</td>
<td>Minimum overlap  to consider a peak as overlapping with a last exon (default: <em>1</em>)</td>
</tr>
<tr class="row-odd"><td>MIN_INTRON_OV</td>
<td>Minimum overlap  to consider a peak as overlapping with a intron (default: <em>3</em>)</td>
</tr>
<tr class="row-even"><td>MIN_ANNOT_OV</td>
<td>Minimum overlap for peaks annotation (default: <em>1</em>)</td>
</tr>
</tbody>
</table>
<hr class="docutils" />
<table border="1" class="docutils">
<colgroup>
<col width="15%" />
<col width="85%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head" colspan="2">SAMPLES COMPARISON</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>COMBINE_SAMPLE</td>
<td>Samples to merge before comparison. Should be under bracket, with comma separated (i.e [1,2,3,4][5,6,7,8] ...)</td>
</tr>
<tr class="row-odd"><td>COMPARE_SAMPLE</td>
<td>Define group to compare. Must be defined as COMBINE_SAMPLE with group 0 vs group 1 (i.e [0,0,1,1][0,0,1,1] ...)</td>
</tr>
<tr class="row-even"><td>MIN_COUNT_PER_COND</td>
<td>Minimum sum of counts per condition. (Default: <em>10</em>)</td>
</tr>
</tbody>
</table>
<hr class="docutils" />
<p>2. Edit the sample list files in case of multiple samples. This file must be tab delimited with <em>sample_number</em> t <em>file</em> t <em>sample_id</em>.
Note that the <em>sample_number</em> must correspond to the COMBINE_SAMPLE variable from the configuration file. These samples will be combined to define a common set of peaks which can further be used for differential analysis.
The <em>sample_id</em> are used for the differential analysis only.
Here is an sample list file example :</p>
<dl class="docutils">
<dt>::</dt>
<dd>1     /data/sample1.bam       COND1
2     /data/sample1.bam       COND1
3     /data/sample1.bam       COND2
...</dd>
</dl>
<ol class="arabic simple" start="3">
<li>Run the pipeline</li>
</ol>
<blockquote>
<div><ul class="simple">
<li>For one file</li>
</ul>
<div class="highlight-guess"><div class="highlight"><pre>/bin/smart -c CONFIG -i INPUT_BAM -o OUTPUT_DIR
</pre></div>
</div>
<ul class="simple">
<li>For a list of file</li>
</ul>
<div class="highlight-guess"><div class="highlight"><pre>/bin/smart -c CONFIG -l INPUT_LIST -o OUTPUT_DIR
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="how-does-smart-work">
<h2>How does SMART work ?<a class="headerlink" href="#how-does-smart-work" title="Permalink to this headline">¶</a></h2>
<p>TODO</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">SMART pipeline Manual</a><ul>
<li><a class="reference internal" href="#what-is-the-smart-pipeline">What is the SMART pipeline ?</a></li>
<li><a class="reference internal" href="#how-to-install-it">How to install it ?</a></li>
<li><a class="reference internal" href="#annotation-files">Annotation Files</a></li>
<li><a class="reference internal" href="#how-to-use-it">How to use it ?</a></li>
<li><a class="reference internal" href="#id1">How to use it ?</a></li>
<li><a class="reference internal" href="#how-does-smart-work">How does SMART work ?</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="QUICKSTART.html"
                        title="previous chapter">SMART Quick Start Guide</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/MANUAL.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="QUICKSTART.html" title="SMART Quick Start Guide"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">SMART 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &copy; Copyright 2015, M. Sejourne, A. Teissander, G. Boldina, M. Dutertre, N. Servant.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.1.
    </div>
  </body>
</html>