% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\usepackage[utf8]{inputenc}
\DeclareUnicodeCharacter{00A0}{\nobreakspace}
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{babel}
\usepackage{times}
\usepackage[Bjarne]{fncychap}
\usepackage{longtable}
\usepackage{sphinx}
\usepackage{multirow}

\addto\captionsenglish{\renewcommand{\figurename}{Fig. }}
\addto\captionsenglish{\renewcommand{\tablename}{Table }}
\floatname{literal-block}{Listing }



\title{SMART pipeline Documentation}
\date{April 22, 2015}
\release{1.0}
\author{M. Sejourne, A. Teissander, G. Boldina, M. Dutertre, N. Servant}
\newcommand{\sphinxlogo}{}
\renewcommand{\releasename}{Release}
\makeindex

\makeatletter
\def\PYG@reset{\let\PYG@it=\relax \let\PYG@bf=\relax%
    \let\PYG@ul=\relax \let\PYG@tc=\relax%
    \let\PYG@bc=\relax \let\PYG@ff=\relax}
\def\PYG@tok#1{\csname PYG@tok@#1\endcsname}
\def\PYG@toks#1+{\ifx\relax#1\empty\else%
    \PYG@tok{#1}\expandafter\PYG@toks\fi}
\def\PYG@do#1{\PYG@bc{\PYG@tc{\PYG@ul{%
    \PYG@it{\PYG@bf{\PYG@ff{#1}}}}}}}
\def\PYG#1#2{\PYG@reset\PYG@toks#1+\relax+\PYG@do{#2}}

\expandafter\def\csname PYG@tok@gd\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.63,0.00,0.00}{##1}}}
\expandafter\def\csname PYG@tok@gu\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.50,0.00,0.50}{##1}}}
\expandafter\def\csname PYG@tok@gt\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.27,0.87}{##1}}}
\expandafter\def\csname PYG@tok@gs\endcsname{\let\PYG@bf=\textbf}
\expandafter\def\csname PYG@tok@gr\endcsname{\def\PYG@tc##1{\textcolor[rgb]{1.00,0.00,0.00}{##1}}}
\expandafter\def\csname PYG@tok@cm\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@vg\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@m\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@mh\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@cs\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}\def\PYG@bc##1{\setlength{\fboxsep}{0pt}\colorbox[rgb]{1.00,0.94,0.94}{\strut ##1}}}
\expandafter\def\csname PYG@tok@ge\endcsname{\let\PYG@it=\textit}
\expandafter\def\csname PYG@tok@vc\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@il\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@go\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.20,0.20,0.20}{##1}}}
\expandafter\def\csname PYG@tok@cp\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@gi\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.63,0.00}{##1}}}
\expandafter\def\csname PYG@tok@gh\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.00,0.50}{##1}}}
\expandafter\def\csname PYG@tok@ni\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.84,0.33,0.22}{##1}}}
\expandafter\def\csname PYG@tok@nl\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.13,0.44}{##1}}}
\expandafter\def\csname PYG@tok@nn\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.05,0.52,0.71}{##1}}}
\expandafter\def\csname PYG@tok@no\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.38,0.68,0.84}{##1}}}
\expandafter\def\csname PYG@tok@na\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@nb\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@nc\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.05,0.52,0.71}{##1}}}
\expandafter\def\csname PYG@tok@nd\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.33,0.33,0.33}{##1}}}
\expandafter\def\csname PYG@tok@ne\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@nf\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.02,0.16,0.49}{##1}}}
\expandafter\def\csname PYG@tok@si\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.44,0.63,0.82}{##1}}}
\expandafter\def\csname PYG@tok@s2\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@vi\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@nt\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.02,0.16,0.45}{##1}}}
\expandafter\def\csname PYG@tok@nv\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@s1\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@gp\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.78,0.36,0.04}{##1}}}
\expandafter\def\csname PYG@tok@sh\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@ow\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@sx\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.78,0.36,0.04}{##1}}}
\expandafter\def\csname PYG@tok@bp\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@c1\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@kc\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@c\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@mf\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@err\endcsname{\def\PYG@bc##1{\setlength{\fboxsep}{0pt}\fcolorbox[rgb]{1.00,0.00,0.00}{1,1,1}{\strut ##1}}}
\expandafter\def\csname PYG@tok@mb\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@ss\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.32,0.47,0.09}{##1}}}
\expandafter\def\csname PYG@tok@sr\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.14,0.33,0.53}{##1}}}
\expandafter\def\csname PYG@tok@mo\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@kd\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@mi\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@kn\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@o\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\expandafter\def\csname PYG@tok@kr\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@s\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@kp\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@w\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.73,0.73}{##1}}}
\expandafter\def\csname PYG@tok@kt\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.56,0.13,0.00}{##1}}}
\expandafter\def\csname PYG@tok@sc\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@sb\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@k\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@se\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@sd\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}

\def\PYGZbs{\char`\\}
\def\PYGZus{\char`\_}
\def\PYGZob{\char`\{}
\def\PYGZcb{\char`\}}
\def\PYGZca{\char`\^}
\def\PYGZam{\char`\&}
\def\PYGZlt{\char`\<}
\def\PYGZgt{\char`\>}
\def\PYGZsh{\char`\#}
\def\PYGZpc{\char`\%}
\def\PYGZdl{\char`\$}
\def\PYGZhy{\char`\-}
\def\PYGZsq{\char`\'}
\def\PYGZdq{\char`\"}
\def\PYGZti{\char`\~}
% for compatibility with earlier versions
\def\PYGZat{@}
\def\PYGZlb{[}
\def\PYGZrb{]}
\makeatother

\renewcommand\PYGZsq{\textquotesingle}

\begin{document}

\maketitle
\tableofcontents
\phantomsection\label{index::doc}


Contents:


\chapter{SMART Quick Start Guide}
\label{QUICKSTART:welcome-to-the-smart-documentation}\label{QUICKSTART::doc}\label{QUICKSTART:smart-quick-start-guide}
See NEWS for information about changes in this and previous versions

See LOGBOOK for details about the HiC-Pro developmens


\section{What is the SMART pipeline ?}
\label{QUICKSTART:what-is-the-smart-pipeline}
SMART (3'-Seq Mapping Annotation and Regulation Tool) was set up to process 3' sequencing data from aligned sequencing reads.
It includes the peak detection, filtering, and annotation.
If you use it, please cite :
\emph{Boldina et al. 2015}


\section{How to install it ?}
\label{QUICKSTART:how-to-install-it}
The following dependancies are required :
*
* R with the \emph{RColorBrewer}, \emph{ggplot2}, \emph{rtracklayer}, \emph{DESeq2}, \emph{magrittr}, \emph{dplyr}, \emph{qplots} and \emph{genomicRanges} packages
* Samtools (\textgreater{}0.1.18)
* BEDTools (\textgreater{}2.17.0)

To install the SMART pipeline, simply extract the archive and set up the configuration file with the paths to dependencies.

\begin{Verbatim}[commandchars=\\\{\}]
tar \PYGZhy{}zxvf smart\PYGZus{}1.0.0.tar.gz
cd smart\PYGZus{}1.0.0
\end{Verbatim}


\section{Annotation Files}
\label{QUICKSTART:annotation-files}
The pipeline is using a couple of annotation files with gene annotation and last exons information. These files are based on UCSC Refseq gene.
In order to generate all required annotation files, please set the ANNOT\_DIR, ORG and UCSC\_EXPORT in the configuration file

\begin{Verbatim}[commandchars=\\\{\}]
BUILD\PYGZus{}ANNOT=1
ORG=mm9
UCSC\PYGZus{}EXPORT=refseq\PYGZus{}export\PYGZus{}mm9.csv
\end{Verbatim}

SMART will start by creating the annotation files in the forder ANNOT\_DIR/ORG/
The annotation only have to be generated once. Mouse annotation are provided with the pipeline.


\section{How to use it ?}
\label{QUICKSTART:how-to-use-it}
SMART can be used both for a single sample or for a list of samples.
In the case of sample list, peaks detected in all samples are merged before annotation.
In oder to use the pipeline, please set up the configuration according to your analysis, and run the following command :

\begin{Verbatim}[commandchars=\\\{\}]
/bin/smart \PYGZhy{}c CONFIG [\PYGZhy{}i INPUT\PYGZus{}BAM] [\PYGZhy{}l INPUT\PYGZus{}LIST] \PYGZhy{}o OUTPUT\PYGZus{}DIR
\end{Verbatim}


\chapter{SMART pipeline Manual}
\label{MANUAL:smart-pipeline-manual}\label{MANUAL::doc}

\section{What is the SMART pipeline ?}
\label{MANUAL:what-is-the-smart-pipeline}
SMART was set up to process PolyA sequencing data from aligned sequencing reads.
It includes the peak detection, filtering, and annotation.
If you use it, please cite :
\emph{Boldina et al. 2015}


\section{How to install it ?}
\label{MANUAL:how-to-install-it}
The following dependancies are required :
*
* R with the \emph{RColorBrewer}, \emph{ggplot2}, \emph{rtracklayer}, and \emph{genomicRanges} packages
* Samtools (\textgreater{}0.1.18)
* BEDTools (\textgreater{}2.17.0)

To install the SMART pipeline, simply extract the archive.

\begin{Verbatim}[commandchars=\\\{\}]
tar \PYGZhy{}zxvf smart\PYGZus{}1.0.0.tar.gz
cd smart\PYGZus{}1.0.0
\end{Verbatim}

Then, edit the \emph{config.txt} file and manually defined the paths to dependencies.

\begin{tabulary}{\linewidth}{|L|L|}
\hline
 \multicolumn{2}{|l|}{\textsf{\relax 
SYSTEM CONFIGURATION
}}\\
\hline
SAMTOOLS\_PATH
 & 
Full path to the samtools installation directory (\textgreater{}0.1.18)
\\
\hline
BEDTOOLS\_PATH
 & 
Full path to the BEDTools installation directory (\textgreater{}2.17.0)
\\
\hline
R\_PATH
 & 
Full path to the R installation directory
\\
\hline
PYTHON\_PATH
 & 
Full path to the python installation directory
\\
\hline
AWK\_PATH
 & 
Full path to the awk utility
\\
\hline\end{tabulary}



\section{Annotation Files}
\label{MANUAL:annotation-files}
The pipeline is using a couple of annotation files with gene annotation and last exons information. These files are based on UCSC Refseq gene.
This file can be downloaded from the \href{https://genome.ucsc.edu/cgi-bin/hgTables}{UCSC TableBrowser website} and should look like this :

\begin{Verbatim}[commandchars=\\\{\}]
\PYGZsh{}bin  name    chrom   strand  txStart txEnd   cdsStart
cdsEnd        exonCount       exonStarts
exonEnds      score   name2   cdsStartStat    cdsEndStat      exonFrames
0     NM\PYGZus{}001195025    chr1    +       134212701       134230065       134212806
      134228958       8       134212701,134221529,134222782,134224273,134224707,
      134226534,134227135,134227897,  134213049,134221650,134222806,134224425,
      134224773,134226654,134227268,134230065,        0       Nuak2   cmpl
      cmpl    0,0,1,1,0,0,0,1,
0     NM\PYGZus{}028778       chr1    +       134212701       134230065       134212806
      134228958       7       134212701,134221529,134224273,134224707,134226534,
      134227135,134227897,    134213049,134221650,134224425,134224773,134226654,
      134227268,134230065,    0       Nuak2   cmpl    cmpl    0,0,1,0,0,0,1,
\end{Verbatim}

In order to generate all required annotation files, please set the ANNOT\_DIR, ORG and UCSC\_EXPORT in the configuration file

\begin{Verbatim}[commandchars=\\\{\}]
BUILD\PYGZus{}ANNOT=1
ORG=mm9
UCSC\PYGZus{}EXPORT=refseq\PYGZus{}export\PYGZus{}mm9.csv
\end{Verbatim}

The pipeline will start by creating the annotation files in the forder ANNOT\_DIR/ORG/
The annotation only have to be generated once. Mouse annotation are provided with the pipeline.

In addition, SMART will look for known polyA signal in the detected peaks and flanking regions.
This list of motif is defined in the \emph{polyA\_signal.csv} file and can be edited.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{n}{AATAAA}
\PYG{n}{ATTAAA}
\PYG{n}{AGTAAA}
\PYG{n}{TATAAA}
\PYG{n}{CATAAA}
\PYG{n}{AAGAAA}
\PYG{n}{GATAAA}
\PYG{n}{AATATA}
\PYG{n}{AATGAA}
\PYG{n}{TTTAAA}
\PYG{n}{AATACA}
\PYG{n}{AAAAAG}
\PYG{n}{ACTAAA}
\PYG{n}{AATAGA}
\end{Verbatim}


\section{How to use it ?}
\label{MANUAL:how-to-use-it}
SMART can be used both for a single sample or for a list of samples.
In the case of sample list, peaks detected in all samples are merged before annotation.
In oder to use the pipeline, please set up the configuration according to your analysis, and run the following command :

\begin{Verbatim}[commandchars=\\\{\}]
/bin/smart \PYGZhy{}c CONFIG [\PYGZhy{}i INPUT\PYGZus{}BAM] [\PYGZhy{}l INPUT\PYGZus{}LIST] \PYGZhy{}o OUTPUT\PYGZus{}DIR
\end{Verbatim}


\section{How to use it ?}
\label{MANUAL:id1}\begin{enumerate}
\item {} 
Copy and edit the configuration file \emph{`config.txt'} in your local folder. The `{[}' options are optional.

\end{enumerate}

\begin{tabulary}{\linewidth}{|L|L|}
\hline
 \multicolumn{2}{|l|}{\textsf{\relax 
GENE ANNOTATIONS
}}\\
\hline
BUILD\_ANNOT
 & 
0/1 - Run the annotation builder
\\
\hline
ORG
 & 
Organism
\\
\hline
UCSC\_EXPORT
 & 
UCSC reference to build the annotation (i.e. refseq\_export\_mm9.csv)
\\
\hline
POLYA\_MOTIF
 & 
List of polyA annotation signal (i.e annotation/polyA\_signal.csv)
\\
\hline\end{tabulary}



\bigskip\hrule{}\bigskip


\begin{tabulary}{\linewidth}{|L|L|}
\hline
 \multicolumn{2}{|l|}{\textsf{\relax 
PEAK DETECTION
}}\\
\hline
MIN\_MAPQ
 & 
Minimum reads mapping quality (default: \emph{20})
\\
\hline
MAX\_DIST\_MERGE
 & 
Maximum distance between reads to be merged as a peak (default: \emph{170})
\\
\hline
MIN\_NB\_READS\_PER\_PEAK
 & 
Minimum number of reads per peads (default: \emph{5}
\\
\hline\end{tabulary}



\bigskip\hrule{}\bigskip


\begin{tabulary}{\linewidth}{|L|L|}
\hline
 \multicolumn{2}{|l|}{\textsf{\relax 
PEAK FILTERING
}}\\
\hline
NB\_STRETCH\_POLYA
 & 
Window length to look for polyA stretch (default: \emph{9})
\\
\hline
MISM
 & 
Number of non-A base allowed in the NB\_STRETCH\_POLYA window (default: \emph{1})
\\
\hline
NB\_STRETCH\_CONSECUTIVE
 & 
Minimum size of A stretch (default: \emph{6})
\\
\hline
WINSIZE\_DOWN
 & 
Window size downstream the peaks (default: \emph{150})
\\
\hline
WINSIZE\_UP
 & 
Window size upstream the peaks (default: \emph{50})
\\
\hline
KEEP\_LE\_PEAKS
 & 
Always keep peaks in gene's last exon (0/1, default: \emph{1})
\\
\hline\end{tabulary}



\bigskip\hrule{}\bigskip


\begin{tabulary}{\linewidth}{|L|L|}
\hline
 \multicolumn{2}{|l|}{\textsf{\relax 
ANNOTATION
}}\\
\hline
MIN\_LE\_OV
 & 
Minimum overlap  to consider a peak as overlapping with a last exon (default: \emph{1})
\\
\hline
MIN\_INTRON\_OV
 & 
Minimum overlap  to consider a peak as overlapping with a intron (default: \emph{3})
\\
\hline
MIN\_ANNOT\_OV
 & 
Minimum overlap for peaks annotation (default: \emph{1})
\\
\hline\end{tabulary}



\bigskip\hrule{}\bigskip


\begin{tabulary}{\linewidth}{|L|L|}
\hline
 \multicolumn{2}{|l|}{\textsf{\relax 
SAMPLES COMPARISON
}}\\
\hline
COMBINE\_SAMPLE
 & 
Samples to merge before comparison. Should be under bracket, with comma separated (i.e {[}1,2,3,4{]}{[}5,6,7,8{]} ...)
\\
\hline
COMPARE\_SAMPLE
 & 
Define group to compare. Must be defined as COMBINE\_SAMPLE with group 0 vs group 1 (i.e {[}0,0,1,1{]}{[}0,0,1,1{]} ...)
\\
\hline
MIN\_COUNT\_PER\_COND
 & 
Minimum sum of counts per condition. (Default: \emph{10})
\\
\hline\end{tabulary}



\bigskip\hrule{}\bigskip


2. Edit the sample list files in case of multiple samples. This file must be tab delimited with \emph{sample\_number} t \emph{file} t \emph{sample\_id}.
Note that the \emph{sample\_number} must correspond to the COMBINE\_SAMPLE variable from the configuration file. These samples will be combined to define a common set of peaks which can further be used for differential analysis.
The \emph{sample\_id} are used for the differential analysis only.
Here is an sample list file example :
\begin{description}
\item[{::}] \leavevmode
1     /data/sample1.bam       COND1
2     /data/sample1.bam       COND1
3     /data/sample1.bam       COND2
...

\end{description}
\begin{enumerate}
\setcounter{enumi}{2}
\item {} 
Run the pipeline

\end{enumerate}
\begin{quote}
\begin{itemize}
\item {} 
For one file

\end{itemize}

\begin{Verbatim}[commandchars=\\\{\}]
/bin/smart \PYGZhy{}c CONFIG \PYGZhy{}i INPUT\PYGZus{}BAM \PYGZhy{}o OUTPUT\PYGZus{}DIR
\end{Verbatim}
\begin{itemize}
\item {} 
For a list of file

\end{itemize}

\begin{Verbatim}[commandchars=\\\{\}]
/bin/smart \PYGZhy{}c CONFIG \PYGZhy{}l INPUT\PYGZus{}LIST \PYGZhy{}o OUTPUT\PYGZus{}DIR
\end{Verbatim}
\end{quote}


\section{How does SMART work ?}
\label{MANUAL:how-does-smart-work}
TODO



\renewcommand{\indexname}{Index}
\printindex
\end{document}
